DOCS-1247-PGD-6.0-Docs (#6763)
EnterpriseDB/docs: product_docs/docs/pgd/6/reference/tables-views-functions/nodes-management-interfaces.mdx

| `default_commit_scope`    | `text`     | The commit scope to use by default, initially the `local` commit scope. This option applies only to the top-level node group. You can use individual rules for different origin groups of the same commit scope. See [Origin groups](../commit-scopes/origin_groups) for more details.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |

> | `enable_routing`    | `boolean`  | Where [`Connection Manager`](/pgd/latest/reference/connection-manager/) through the group write leader is enabled for given group. Valid values are `on` or `off`. Default is `on` for subgroups and off for the cluster group.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |

Reviewer: ```suggestion
| `enable_routing`    | `boolean`  | Where [`Connection Manager`](/pgd/latest/reference/connection-manager/) through the group write leader is enabled for a given group. Valid values are `on` or `off`. Default is `on` for subgroups and `off` for the cluster group.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |
```

If `wait_for_completion` is specified as `false`, the function call returns
as soon as the joining procedure starts. You can see the progress of the join in
the log files and the [`bdr.event_summary`](/pgd/latest/reference/tables-views-functions/catalogs-internal#bdrevent_summary)
information view. You can call the function [`bdr.wait_for_join_completion()`](#bdrwait_for_join_completion)
after `bdr.join_node_group()` to wait for the join operation to complete.
It can emit progress information if [`bdr.wait_for_join_completion()`](#bdrwait_for_join_completion) is called with `verbose_progress` set to `true`.

### Notes

This function passes a request to the group consensus mechanism by way of the node
that the `join_target_dsn` connection string points to.
The changes made are replicated globally by the consensus mechanism.

> The function isn't transactional, and will emit an ERROR if executed in a transaction.

Reviewer: ```suggestion
The function isn't transactional and will emit an error if executed in a transaction.
```

The joining process happens in the background and you can't roll it back.
The changes are visible only to the local session if `wait_for_completion` is set to `true` or by calling `bdr.wait_for_join_completion` later.

A node can be part of only a single group, so you can call this function only once
on each node.

Node join doesn't hold any locks in the PGD group.

## `bdr.part_node`

> Removes (parts) the node from the PGD group and eventually removes the parted node’s  metadata automatically from all nodes in the cluster.

Reviewer: ```suggestion
Removes (parts) the node from the PGD group and eventually removes the parted node’s metadata from all nodes in the cluster.
```

> * For the local node, it will remove all the node metadata, including information about remote nodes.

Reviewer: ```suggestion
* For the local node, it removes all the node metadata, including information about remote nodes.
```

* For remote nodes, it removes only the metadata for that specific node.

> This operation does not remove data from the node.

Reviewer: ```suggestion
This operation doesn't remove data from the node.
```

The local node changes membership from its current subgroup to another subgroup in the same PGD node group without needing to part the cluster.
The node's kind must match that of existing nodes in the target subgroup.

Node switching doesn't hold any locks in the PGD group.

Restrictions: currently, the function allows switching only between a subgroup and its PGD node group. To effect a move between subgroups you need to make two separate calls: 1) switch from subgroup to node group and, 2) switch from node group to other subgroup.

## `bdr.sync_node_cancel`

Cancels a sync request for the specified origin and source nodes.

### Synopsis

```sql
bdr.sync_node_cancel(origin text, source text)
```

### Parameters

| Name    | Description                                                                 |
|---------|-----------------------------------------------------------------------------|
| `origin` | Name of the origin node.                                                    |
| `source` | Name of the source node.                                                    |

### Notes

This function cancels all sync node requests for all targets that have the given origin and source.

> This can be invoked only from a write lead.

Reviewer: ```suggestion
You can invoke it only from a write lead.
```
